codegen-maven-plugin/src/main/java/software/amazon/awssdk/codegen/maven/plugin/GenerationMojo.java

```diff
@@ -76,22 +84,59 @@ public void execute() throws MojoExecutionException {
         this.resourcesDirectory = Paths.get(outputDirectory).resolve("generated-resources").resolve("sdk-resources");
         this.testsDirectory = Paths.get(outputDirectory).resolve("generated-test-sources").resolve("sdk-tests");
 
-        findModelRoots().forEach(p -> {
-            Path modelRootPath = p.modelRoot;
-            getLog().info("Loading from: " + modelRootPath.toString());
-            generateCode(C2jModels.builder()
-                                  .customizationConfig(p.customizationConfig)
-                                  .serviceModel(loadServiceModel(modelRootPath))
-                                  .waitersModel(loadWaiterModel(modelRootPath))
-                                  .paginatorsModel(loadPaginatorModel(modelRootPath))
-                                  .endpointRuleSetModel(loadEndpointRuleSetModel(modelRootPath))
-                                  .endpointTestSuiteModel(loadEndpointTestSuiteModel(modelRootPath))
-                                  .build());
+        List<GenerationParams> generationParams = initGenerationParams();
+
+        Map<String, IntermediateModel> serviceNameToModelMap = new HashMap<>();
+
+        generationParams.forEach(
+            params -> {
+                IntermediateModel model = params.intermediateModel;
+                String lowercaseServiceName = StringUtils.lowerCase(model.getMetadata().getServiceName());
+                IntermediateModel previous = serviceNameToModelMap.put(lowercaseServiceName, model);
+                if (previous != null) {
+                    String warning = String.format("Multiple service models found with service name %s. Model validation "
+                                                   + "will likely be incorrect", lowercaseServiceName);
```

Comment:
Why is `will likely be incorrect` do we have any valid cases where multiple models have same service name , why just warn and not error ?

Reply:
I can't think of a valid case honestly. I only encountered this in testing when the generator runs for the `codegen-generated-classes-test` since we have a lot of test models there and some have the same name.